src/Symfony/Component/Security/Core/Authorization/Voter/TraceableVoter.php

```diff
@@ -33,9 +33,9 @@ public function __construct(VoterInterface $voter, EventDispatcherInterface $eve
         $this->eventDispatcher = $eventDispatcher;
     }
 
-    public function vote(TokenInterface $token, $subject, array $attributes): int
+    public function vote(TokenInterface $token, $subject, array $attributes)
     {
-        $result = $this->voter->vote($token, $subject, $attributes);
+        $result = \is_int($vote = $this->voter->vote($token, $subject, $attributes)) ? Vote::create($vote) : $vote;
```

Comment:
Maybe trigger a deprecation here when the result is an integer?

Reply:
Maybe you should also update the VoterInterface::vote() description here
https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Security/Core/Authorization/Voter/VoterInterface.php#L30L31